pkg/net/net.go

```diff
@@ -74,6 +75,61 @@ func (o *Overlay) loop(ch <-chan EvtState, state event.Emitter) {
 			}
 		}
 	}()
+
+	for {
+		time.Sleep(500 * time.Millisecond) // TODO: decide optimal tick for updating neighbors
+		peer, err := o.n.RandPeer()
+		if err != nil { // TODO: decide what to do errors
+			continue
+		}
+		peers, err := gossip{
+			o.n, o.h, o.h, peer, protoutil.Join(o.proto, "gossip"),
+		}.PushPull(o.ctx())
+		if err != nil {
+			continue
+		}
+		err = o.setNeighborhood(o.ctx(), peers)
+		if err != nil {
+			continue
+		}
+	}
+}
+
+func (o *Overlay) setNeighborhood(ctx context.Context, peers []*peer.PeerRecord) error {
+	// get new nodes
+	oldPeers := o.n.Peers()
+	newPeers := make(StaticAddrs, len(peers))
+	for _, p := range peers {
+		if !contains(oldPeers, p) {
+			newPeers = append(newPeers, peer.AddrInfo{p.PeerID, p.Addrs})
+		}
+	}
+	// get evict nodes
+	evictPeers := []*peer.PeerRecord{}
+	for _, p := range oldPeers {
+		if !contains(peers, p) {
+			evictPeers = append(evictPeers, p)
+		}
+	}
+	// join new nodes
+	err := o.Join(ctx, newPeers)
+	if err != nil {
+		return err
+	}
+	// evict old unused nodes
+	for _, p := range evictPeers {
+		o.n.Evict(ctx, p.PeerID)
+	}
+	return nil
+}
+
+func contains(peers []*peer.PeerRecord, rec *peer.PeerRecord) bool {
```

Comment:
Just a thought:  perhaps we should define a special type for a `[]*peer.PeerRecord`?

If we follow through on my suggestion to use `PeerRecord` directly in `vertex`, then it would be useful to have a type that satisfies `sort.Interface`, so that we can easily order the peers by their hop count.  We could then add `Contains` as a method to this type as well.

Reply:
I agree. Do you have any suggestion for the type name?

Reply:
`recordSlice` ?

Reply:
It's good for me 👍